demo/App.js

```diff
@@ -1,12 +1,177 @@
+
 
 export default class App extends React.Component {
   state = {
     isLoadingComplete: false
   };
 
+
+	componentDidMount() {
+
+		let ropstenProvider = ethers.getDefaultProvider('ropsten');
+		let mainnetProvider = ethers.getDefaultProvider();
+		let utils = ethers.utils;
+
+		let privateKey = "0x3141592653589793238462643383279502884197169399375105820974944592";
+		let address = "0x7357589f8e367c2C31F51242fB77B350A11830F3";
+		let wallet = new ethers.Wallet(privateKey);
+		let fundedWallet = new ethers.Wallet("0x18bfdd05c1e63e4a27081352e2910c164a4d34588f8d7eecfdfcb654742934c2", ropstenProvider); //addr:0xa9adeF68CEF4EA3C9d0717126e14fDfebFDFB4d7
+
+		let wallet2mnemonic = "radar blur cabbage chef fix engine embark joy scheme fiction master release";
+		let wallet2 = ethers.Wallet.fromMnemonic(wallet2mnemonic);
+		let wallet3 = ethers.Wallet.createRandom();
+
+		console.assert(wallet.address === address, "wallet.address");
+		console.assert(wallet.privateKey === privateKey, "wallet.privateKey");
+		console.assert(wallet2.mnemonic === wallet2mnemonic, "wallet.privateKey");
+		console.assert(wallet3.address !== undefined, "wallet.address from randonly created");
+
+		/*** Scrypt encryption is VERY SLOW on react-native ***/
+		/* import wallet from json */
+		// let data = {
+		//     id: "fb1280c0-d646-4e40-9550-7026b1be504a",
+		//     address: "88a5c2d9919e46f883eb62f7b8dd9d0cc45bc290",
+		//     Crypto: {
+		//         kdfparams: {
+		//             dklen: 32,
+		//             p: 1,
+		//             salt: "bbfa53547e3e3bfcc9786a2cbef8504a5031d82734ecef02153e29daeed658fd",
+		//             r: 8,
+		//             n: 262144
+		//         },
+		//         kdf: "scrypt",
+		//         ciphertext: "10adcc8bcaf49474c6710460e0dc974331f71ee4c7baa7314b4a23d25fd6c406",
+		//         mac: "1cf53b5ae8d75f8c037b453e7c3c61b010225d916768a6b145adf5cf9cb3a703",
+		//         cipher: "aes-128-ctr",
+		//         cipherparams: {
+		//             iv: "1dcdf13e49cea706994ed38804f6d171"
+		//          }
+		//     },
+		//     "version" : 3
+		// };
+		//
+		// let json = JSON.stringify(data);
+		// let password = "foo";
+		//
+		// ethers.Wallet.fromEncryptedWallet(json, password).then(function(wallet) {
+		//     console.log("Address: " + wallet.address);
+		// });
+		//
+		// /* export wallet to JSON */
+		// let password = "password123";
+		// function callback(progress) {
+		//     console.log("Encrypting: " + parseInt(progress * 100) + "% complete");
+		// }
+		// wallet.encrypt(password, callback).then(function(json) {
+		//     console.log(json);
+		// });
+
+		// All properties are optional
+		let transaction = {
+			nonce: 0,
+			gasLimit: 21000,
+			gasPrice: ethers.utils.bigNumberify("20000000000"),
+			to: "0x88a5C2d9919e46F883EB62F7b8Dd9d0CC45bc290",
+			value: ethers.utils.parseEther("1.0"),
+			data: "0x",
+			chainId: ethers.utils.getNetwork('homestead').chainId
+		}
+
+		fundedWallet.sign(transaction).then((signedTransaction) => {
```

Comment:
@marcelomorgado Let's change this to async/await before merging, if that works for you

Reply:
Sure